provider/pkg/resources/checkpoint.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2024, Pulumi Corporation.
+
+package resources
+
+import (
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+)
+
+// CheckpointObject puts inputs in the `__inputs` field of the state.
```

Comment:
Out of scope here but I am curious now do we understand why it's needed? Throwing a sentence on why we're doing this could be interesting.

Reply:
This was done to make diff/read more robust (https://github.com/pulumi/pulumi-aws-native/pull/17) and solve this issue: https://github.com/pulumi/pulumi-aws-native/issues/16.

In detail, having access to the old inputs during Read.
I'll put this in the docs in a separate PR